brayns/utils/DynamicLib.cpp

```diff
@@ -18,96 +18,54 @@
  * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#include <algorithm>
 #include <stdexcept>
+#include <string>
 
-#ifdef _WIN32
-    #ifndef WIN32_LEAN_AND_MEAN
-        #define WIN32_LEAN_AND_MEAN
-    #endif
-    #include <windows.h>
-#else
-    #include <dlfcn.h>
-    #include <sys/times.h>
-#endif
+#include <dlfcn.h>
+#include <sys/times.h>
 
 #include "DynamicLib.h"
 
 namespace brayns
 {
-DynamicLib::DynamicLib(const std::string &name)
+DynamicLib::DynamicLib(std::string_view name)
 {
-    std::string file = name;
-    std::string errorMessage;
-#ifdef _WIN32
-    std::string filename = name + ".dll";
-    _handler = LoadLibrary(fullName.c_str());
-    if (!_handler)
+    auto filename = "lib" + std::string(name) + ".so";
+    _handle = dlopen(filename.c_str(), RTLD_LAZY | RTLD_GLOBAL);
+    if (!_handle)
     {
-        DWORD err = GetLastError();
-        LPTSTR buffer;
-        FormatMessage(
-            FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS,
-            NULL,
-            err,
-            MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT),
-            (LPTSTR)&buffer,
-            0,
-            NULL);
-        errorMessage = buffer;
-        LocalFree(buffer);
-    }
-
-#else
-    #if defined(__MACOSX__) || defined(__APPLE__)
-    std::string filename = "lib" + file + ".dylib";
-    #else
-    std::string filename = "lib" + file + ".so";
-    #endif
-    _handler = dlopen(filename.c_str(), RTLD_LAZY | RTLD_GLOBAL);
-    if (!_handler)
-        errorMessage = dlerror();
-#endif
-    if (!_handler)
+        std::string errorMessage = dlerror();
         throw std::runtime_error("Error opening dynamic library: " + filename + ": " + errorMessage);
+    }
 }
 
 DynamicLib::~DynamicLib()
 {
-    if (!_handler)
+    if (!_handle || _handle == RTLD_DEFAULT)
+    {
         return;
-
-#ifdef _WIN32
-    if (_handler != GetModuleHandle(0))
-        FreeLibrary(_handler);
-#else
-    if (_handler != RTLD_DEFAULT)
-        dlclose(_handler);
-#endif
-    _handler = 0;
+    }
+    dlclose(_handle);
 }
 
 DynamicLib::DynamicLib(DynamicLib &&other)
 {
-    _handler = other._handler;
-    other._handler = nullptr;
+    _handle = std::exchange(other._handle, nullptr);
 }
 
 DynamicLib &DynamicLib::operator=(DynamicLib &&other)
 {
-    _handler = other._handler;
-    other._handler = nullptr;
+    std::swap(_handle, other._handle);
```

Comment:
Why swap here and exchange in the constructor?
Personal preference for me is swap, since exchange is, at best, as fast as swap.